client/state/reader/start/reducer.js

```diff
@@ -21,19 +24,30 @@ import { isValidStateWithSchema } from 'state/utils';
 /**
  * Tracks all known list objects, indexed by list ID.
  *
- * @param  {Object} state  Current state
+ * @param  {Array} state  Current state
  * @param  {Object} action Action payload
- * @return {Object}        Updated state
+ * @return {Array}        Updated state
  */
-export function items( state = {}, action ) {
+export function items( state = [], action ) {
 	switch ( action.type ) {
 		case READER_START_RECOMMENDATIONS_RECEIVE:
-			return Object.assign( {}, state, keyBy( action.recommendations, 'ID' ) );
+			// Filter out any recommendations we already have
+			const newRecommendations = filter( action.recommendations, ( incomingRecommendation ) => {
+				if ( findIndex( state, ( existingRecommendation ) => {
```

Comment:
i think this can just be `return ! find( state, { ID: incomingRecommendation.ID } )` if you use `lodash/find`?  

if you don't like the negative in the boolean you could use `lodash/reject` instead. `reject( state, { ID: incomingRecommendation.ID } )`